Tests/swift-sdk-swift-tests/IterableAutoRegistrationTests.swift

```diff
@@ -0,0 +1,207 @@
+//
+//
+//  Created by Tapash Majumder on 8/15/18.
+//  Copyright © 2018 Iterable. All rights reserved.
+//
+
+import XCTest
+
+@testable import IterableSDK
+
+class IterableAutoRegistrationTests: XCTestCase {
+    private static let apiKey = "zeeApiKey"
+
+    override func setUp() {
+        super.setUp()
+
+        TestUtils.clearUserDefaults()
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    func testCallDisableAndEnable() {
+        let expectation1 = XCTestExpectation(description: "call register device API")
+        let expectation2 = XCTestExpectation(description: "call registerForRemoteNotifications twice")
+        let expectation3 = XCTestExpectation(description: "call disable on user1@example.com")
+        
+        let networkSession = MockNetworkSession(statusCode: 200)
+        let config = IterableConfig()
+        config.pushIntegrationName = "my-push-integration"
+        config.autoPushRegistration = true
+        
+        let notificationStateProvider = MockNotificationStateProvider(enabled: true)
+        var registerForRemoteNotificationsCount = 0
+        notificationStateProvider.callback = {
+            registerForRemoteNotificationsCount += 1
+            
+            if registerForRemoteNotificationsCount == 2 {
+                expectation2.fulfill()
+            }
+        }
+        
+        IterableAPI.initialize(apiKey: IterableAutoRegistrationTests.apiKey, config:config, networkSession: networkSession, notificationStateProvider: notificationStateProvider)
+        IterableAPI.email = "user1@example.com"
+        let token = "zeeToken".data(using: .utf8)!
+        networkSession.callback = {(_, _, _) in
```

Comment:
I think it's generally better to store requests in an array in the mock network session and then verify their content, that way the control flow would be easier to understand.

Reply:
I like the idea. I will do that when I work on codecov part 3.